src/bicycle/model.py

```diff
@@ -1,18 +1,55 @@
+"""
+This module implements the Bicycle model for gene regulatory network (GRN) inference.
+
+The algorithm is based on the paper "Bicycle: Intervention-Based Causal Discovery with Cycles"
+by Rohbeck et al. It provides a PyTorch Lightning implementation of the Bicycle model, which is
+designed to infer causal relationships in GRNs using perturbation-based data.
+
+Classes:
+    - Encoder: Encodes input data into a latent space representation for use in a
+      Variational Autoencoder (VAE) or similar architecture.
+    - Omega_Iterative: Optimizes a matrix `omega` using Lyapunov stability principles
+      for steady-state assumptions in GRNs.
+    - BICYCLE: The main model class that integrates various components for GRN inference,
+      including latent variable modeling, intervention handling, and loss computation.
+
+Features:
+    - Supports multiple loss functions, including KL divergence, negative log-likelihood (NLL),
+      Lyapunov loss, and sparsity loss.
+    - Handles various intervention types, including CRISPR-based perturbations (e.g., Cas9, dCas9).
+    - Provides options for latent variable modeling using an encoder or direct parameterization.
+    - Implements spectral loss to ensure stability in gene-gene interaction matrices.
```

Comment:
Can this be combined with the first bullet point (which already addresses the losses)?